src/main/java/dev/the_fireplace/unforgivingvoid/usecase/SpawnPositionLocator.java

```diff
@@ -122,23 +115,39 @@ public BlockPos findSpawnPosition(EntityType<?> entityType, ServerWorld targetWo
                 return targetFocalPosition;
             }
 
-            int targetX = applyHorizontalOffset(rand, targetFocalPosition.getX());
-            int targetZ = applyHorizontalOffset(rand, targetFocalPosition.getZ());
-            int targetY = targetWorld.getTopY(Heightmap.Type.WORLD_SURFACE, targetX, targetZ);
-            BlockPos attemptPos = new BlockPos(targetX, targetY, targetZ);
-
-            spawnVec = findSafePlatform(entityType, targetWorld, attemptPos);
+            spawnVec = findSafePlatform(entityType, targetWorld, targetFocalPosition);
         }
 
         return new BlockPos(spawnVec.get());
     }
 
-    private Optional<Vec3d> findSafePlatform(EntityType<?> entityType, ServerWorld targetWorld, BlockPos blockPos) {
-        return safePosition.findBy(entityType, targetWorld, blockPos);
+    private Optional<Vec3d> findSafePlatform(EntityType<?> entityType, ServerWorld targetWorld, BlockPos targetFocalPosition, int targetY) {
+        Random rand = targetWorld.getRandom();
+
+        int targetX = applyHorizontalOffset(rand, targetFocalPosition.getX());
+        int targetZ = applyHorizontalOffset(rand, targetFocalPosition.getZ());
+
+        BlockPos attemptPos = new BlockPos(targetX, targetY, targetZ);
+
+        return safePosition.findBy(entityType, targetWorld, attemptPos);
+    }
+
+
+    private Optional<Vec3d> findSafePlatform(EntityType<?> entityType, ServerWorld targetWorld, BlockPos targetFocalPosition) {
+        Random rand = targetWorld.getRandom();
+
+        int targetX = applyHorizontalOffset(rand, targetFocalPosition.getX());
+        int targetZ = applyHorizontalOffset(rand, targetFocalPosition.getZ());
+        int targetY = targetWorld.getTopY(Heightmap.Type.WORLD_SURFACE, targetX, targetZ);
+        BlockPos attemptPos = new BlockPos(targetX, targetY, targetZ);
+
+        return safePosition.findBy(entityType, targetWorld, attemptPos);
     }
```

Comment:
We can reuse the other findSafePlatform to reduce the amount of repeated code here.
```suggestion
    private Optional<Vec3d> findSafePlatform(EntityType<?> entityType, ServerWorld targetWorld, BlockPos targetFocalPosition) {
        int targetY = targetWorld.getTopY(Heightmap.Type.WORLD_SURFACE, targetX, targetZ);

        return findSafePlatform(entityType, targetWorld, targetFocalPosition, targetY);
    }
```

Reply:
This does not work, as `targetWorld.getTopY(Heightmap.Type.WORLD_SURFACE, targetX, targetZ);` requires `targetX` and `targetZ` to be computed.

Reply:
Ah right. In that case we should be good.